test-d/kebab-case.ts

```diff
@@ -0,0 +1,34 @@
+import {KebabCase} from '../source/kebab-case';
+import {expectType, expectAssignable} from 'tsd';
+
+const kebabFromCamel: KebabCase<'fooBar'> = 'foo-bar';
+expectType<'foo-bar'>(kebabFromCamel);
+
+const kebabFromKebab: KebabCase<'foo-bar'> = 'foo-bar';
+expectType<'foo-bar'>(kebabFromKebab);
+
+const kebabFromSpace: KebabCase<'foo bar'> = 'foo-bar';
+expectType<'foo-bar'>(kebabFromSpace);
+
+const kebabFromSnake: KebabCase<'foo_bar'> = 'foo-bar';
+expectType<'foo-bar'>(kebabFromSnake);
+
+const noKebabFromMono: KebabCase<'foobar'> = 'foobar';
+expectType<'foobar'>(noKebabFromMono);
+
+// Verifying example
+type KebabCasedProps<T> = {
+    [K in keyof T as KebabCase<K>]: T[K]
+};
+
+interface CliOptions {
+		dryRun: boolean;
+		includeFile: string;
+    foo: number;
```

Comment:
Indentation is odd here and in other places as well. You should only use tabs.

Hint: if you're using VSCode, install the editorconfig extension and run `npm i -g editorconfig` once and this will become much easier.

Reply:
@papb Thanks for pointing that out 👍 I do have the editorconfig extension installed, however this code was pasted from a place where it had space indention and I unfortunately hadn't turned on whitespace visualization in VS Code for some reason.

Fixed and pushed 👍